nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestAnnotationNotification.java

```diff
@@ -170,38 +178,63 @@ protected void captureAndVerifyAnnotationEventNotification(Supplier<DocumentMode
     @Test
     public void shouldNotifyWithTheRightAnnotatedDocument() {
         // First comment
-        Annotation createdAnnotation = createAnnotation(annotatedDocumentModel);
+        Annotation createdAnnotation = createAnnotation(annotatedDocumentModel, ADMINISTRATOR, ANY_ANNOTATION_MESSAGE);
         DocumentModel createdAnnotationDocModel = session.getDocument(new IdRef(createdAnnotation.getId()));
         // before subscribing, or previous event will be notified as well
         transactionalFeature.nextTransaction();
         // Reply
         captureAndVerifyAnnotationEventNotification(() -> {
             // subscribe to notifications
-            addSubscriptions("CommentAdded");
+            addSubscriptions(COMMENT_ADDED_NOTIFICATION);
 
-            Comment reply = createAnnotation(createdAnnotationDocModel);
+            Comment reply = createAnnotation(createdAnnotationDocModel, ADMINISTRATOR, ANY_ANNOTATION_MESSAGE);
             DocumentModel replyDocumentModel = session.getDocument(new IdRef(reply.getId()));
             return session.getDocument(new IdRef(replyDocumentModel.getId()));
         }, COMMENT_ADDED, DOCUMENT_CREATED);
     }
 
+    @Test
+    @Deploy("org.nuxeo.ecm.platform.comment:OSGI-INF/notification-subscription-contrib.xml")
+    public void testAutoSubscribingOnlyOnceToNewAnnotations() {
+        String john = "john";
+        String johnSubscription = NotificationConstants.USER_PREFIX + john;
+        createUser(john);
+        List<String> subscriptions = notificationManager.getSubscriptionsForUserOnDocument(johnSubscription,
+                annotatedDocumentModel);
+        assertEquals(0, subscriptions.size());
+        createAnnotation(annotatedDocumentModel, john, "Test message");
+        transactionalFeature.nextTransaction();
+        annotatedDocumentModel = session.getDocument(annotatedDocumentModel.getRef());
+        subscriptions = notificationManager.getSubscriptionsForUserOnDocument(johnSubscription, annotatedDocumentModel);
+        List<String> expectedSubscriptions = List.of(COMMENT_ADDED_NOTIFICATION, COMMENT_UPDATED_NOTIFICATION);
+        assertEquals(expectedSubscriptions.size(), subscriptions.size());
+        assertTrue(subscriptions.containsAll(expectedSubscriptions));
```

Comment:
the two assertion can be replaced by one:

```
assertEquals(expectedSubscriptions, subscriptions)
```
they should be sorted

Reply:
This doesn't work. This is because the list is created by looping on the entries of a Set. So the list keeps the order in which they come from the set which is random.
That's why I get updated before added.

So we can either sort thing in the lists, or stick with a contains all which is already in place.

Reply:
Stick with `contains`.